File: algorithms/searching/binary_search/binary_search.py
def binary_search(arr, value):
    left_index = 0
    right_index = len(arr) - 1
    
    while left_index <= right_index:
        middle_index = (left_index + right_index) // 2
        middle_value = arr[middle_index]
        if middle_value == value:
            return middle_index
        elif middle_value > value:
            right_index = middle_index - 1
        else:
            left_index = middle_index + 1

    return -1

File: algorithms/searching/binary_search/binary_search.js
function binarySearch(arr, value) {
    leftIndex = 0;
    rightIndex = arr.length - 1;
    
    while (leftIndex <= rightIndex) {
        middleIndex = Math.floor((leftIndex + rightIndex) / 2);
        middleValue = arr[middleIndex];
        
        if (middleValue === value) {
            return middleIndex;
        }
        else if (middleValue > value) {
            rightIndex = middleIndex - 1;
        }
        else {
            leftIndex = middleIndex + 1;
        }
    }

    return -1;
}